File: app/src/main/java/de/exercicse/jrossbach/podcast/network/PodcastApi.java
package de.exercicse.jrossbach.podcast.network;


import de.exercicse.jrossbach.podcast.network.model.PodcastChannelResponse;
import io.reactivex.Single;
import retrofit2.http.GET;

public interface PodcastApi {


    @GET("/")
    Single<PodcastChannelResponse>getPodcastChannelResponse();
}

File: app/src/main/java/de/exercicse/jrossbach/podcast/network/PodcastApi.kt
package de.exercicse.jrossbach.podcast.network

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface PodcastApi {

    @GET("search")
    fun getPodCastSearchResponse(@Query("q") query: String): Single<PodCastSearchResponse>
}
